model/patch_lifecycle.go

```diff
@@ -491,11 +492,29 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 	if err != nil {
 		return nil, errors.Wrap(err, "fetching project options for patch")
 	}
-	intermediateProject, err := LoadProjectInto(ctx, []byte(p.PatchedParserProject), opts, p.Project, project)
-	if err != nil {
-		return nil, errors.Wrapf(err,
-			"marshalling patched parser project from repository revision '%s'",
-			p.Githash)
+
+	// It used to be that the parser project was not stored until patches were
+	// finalized. Newer patches always store the parser project when the patch
+	// is created rather than when it's finalized. For backward compatibility
+	// with existing unfinalized patches, the parser project must be inserted
+	// now if an old patch is being finalized.
+	mustInsertParserProjectDuringFinalization := p.PatchedParserProject != ""
+	var intermediateProject *ParserProject
+	// TODO (EVG-18700): this if-else most likely is equivalent to just calling
```

Comment:
At what point will it be safe to remove this logic? Is the plan to keep it around for a bit of time for the case where a user created a patch and then comes back months later to finalize it? 
Or is this only meant to stay in prod just to bridge the gap of patches created before the deploy and finalized after?